### PR TITLE
fix(ext/node): fix dns.lookup result ordering

### PR DESCRIPTION
**Pull request by kt3k**
partially unblocks #25470

This PR aligns the resolution of `localhost` hostname to Node.js behavior.

In Node.js `dns.lookup("localhost", (_, addr) => console.log(addr))` prints ipv6 address `::1`, but it prints ipv4 address `127.0.0.1` in Deno. That difference causes some errors in the work of enabling `createConnection` option in `http.request` (#25470). This PR fixes the issue by aligning `dns.lookup` behavior to Node.js.

This PR also changes the following behaviors (resolving TODOs):
- `http.createServer` now listens on ipv6 address `[::]` by default on linux/mac
- `net.createServer` now listens on ipv6 address `[::]` by default on linux/mac

These changes are also alignments to Node.js behaviors.

ref: https://github.com/nodejs/node/pull/39987

### CODE REVIEW

File: tests/unit_node/http_test.ts [hunk 1]
```diff
@@ -314,7 +314,7 @@ Deno.test("[node/http] IncomingRequest socket has remoteAddress + remotePort", a
     remotePort = req.socket.remotePort;
     res.end();
   });
-  server.listen(async () => {
+  server.listen({ host: "0.0.0.0" }, async () => {
```
Comment (satyarohith) [inline tests/unit_node/http_test.ts:317]: Can we remove the host and update the request instead to make sure ipv6 is working as expected?

Comment (kt3k) [reply tests/unit_node/http_test.ts]: Sure. In that case we need to update assertion part instead, as `server.listen` (without host specified) listens on ipv6 (`::1`) on unix and ipv4 (`0.0.0.0`) on windows. See https://github.com/denoland/deno/pull/26264/commits/7def448cf0fab5268d38dfd6381db526652bba8e

**Review by satyarohith — APPROVED** [review]
LGTM

**Review by bartlomieju — COMMENTED** [review]
LGTM!

**Pull request merged**